lib/transformers/add_state.ex

```diff
@@ -21,23 +21,21 @@ defmodule AshStateMachine.Transformers.AddState do
 
     case Ash.Resource.Info.attribute(dsl_state, attribute_name) do
       nil ->
-        default =
-          case AshStateMachine.Info.state_machine_default_initial_state(dsl_state) do
-            {:ok, value} ->
-              value
-
-            _ ->
-              nil
-          end
+        case AshStateMachine.Info.state_machine_default_initial_state(dsl_state) do
+          {:ok, default} ->
+            Ash.Resource.Builder.add_attribute(dsl_state, attribute_name, :atom,
+              default: default,
+              allow_nil?: false,
+              public?: true,
+              constraints: [
+                one_of: all_states
+              ]
+            )
 
-        Ash.Resource.Builder.add_attribute(dsl_state, attribute_name, :atom,
-          default: default,
-          allow_nil?: false,
-          public?: true,
-          constraints: [
-            one_of: all_states
-          ]
-        )
+          _ ->
+            {:error,
+             "default_initial_state must be set for AshStateMachine to be able to add the #{attribute_name} attribute"}
```

Comment:
This made sense to me when I added it. But now that I saw the test fail, I guess you could not have a default to force the passing of the initial state from the outisde. 

So I guess, reverting that commit is the way to go. WDYT?

Reply:
This kinda applies to the whole PR though. While I personally think it doesn't make a lot of sense not to have the default if there is only one option, one could argue it somehow.